### PR TITLE
Implement hook abord on amend

### PR DESCRIPTION
**Pull request by Jeremie-Chauvel**
## Description

I implemented interruption of the hook on `git commit --amend` while investigation `yarn version` command. I recommand reading the linked issue for a more in depth summary of why `yarn release` is not compatible with gitmoji

Issue: linked to https://github.com/carloscuesta/gitmoji-cli/issues/497

## Tests

- [x] All tests passed.
- I added a test and fixed previous ones

ps: love the migration to github actions :heart:

### CODE REVIEW

File: src/commands/hook/hook.js [hunk 1]
```diff
@@ -4,7 +4,7 @@ const HOOK: Object = {
   PATH: '/hooks/prepare-commit-msg',
   CONTENTS:
     '#!/bin/sh\n# gitmoji as a commit hook\n' +
-    'exec < /dev/tty\ngitmoji --hook $1\n'
+    'exec < /dev/tty\ngitmoji --hook $1 $2\n'
```
Comment (carloscuesta) [inline src/commands/hook/hook.js:7]: Why we need to pass a second arg?

**Review by carloscuesta — COMMENTED** [review]
Just a small question!

Thanks for the pr ❤️

File: src/commands/hook/hook.js (hunk 1 above)
Comment (Jeremie-Chauvel) [reply src/commands/hook/hook.js:7]: This second argument is there to access the [commit message source from git](https://git-scm.com/docs/githooks#_prepare_commit_msg), allowing gitmoji-cli to detect amending, merge, etc.

**Review by carloscuesta — APPROVED** [review]
Let's merge this

Thanks <3

**Pull request merged**